packager/certcache/certcache.go

```diff
@@ -122,15 +143,40 @@ func (this *CertCache) Init(stop chan struct{}) error {
 	//    to raise an alert if something has gone really wrong.
 	// 7. The ability to serve old responses while fetching new responses.
 	go this.maintainOCSP(stop)
+
+	if this.certFetcher != nil {
+		// Update Certs in the background.
+		go this.maintainCerts(stop)
+	}
+
+	this.isInitialized = true
 	return nil
 }
 
 // For now this just returns the first entry in the certs field in the cache.
 // For follow-on changes, we will transform this to a lambda so that anything
 // that needs a cert can do the cert refresh logic (if needed)  on demand.
 func (this *CertCache) GetLatestCert() (*x509.Certificate) {
-	// TODO(banaag): check if cert is still valid, refresh if not.
-	return this.certs[0]
+	if !this.isInitialized || this.certFetcher == nil {
+		// If certcache is not initialized or certFetcher is not set,
+		// just return cert without checking if it needs auto-renewal.
+		return this.certs[0]
+	}
+	d, err := util.GetDurationToExpiry(this.certs[0], this.certs[0].NotAfter)
+	if err != nil {
+		// Current cert is already invalid. Check if renewal is available.
```

Comment:
Consider using TODO to mark places for the follow-on changes.

Reply:
I already try to do this, what you pointed out is code I missed updating and now fixed. Thanks!